src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -1719,37 +1719,53 @@ def list_vm_images(cmd, image_location=None, publisher_name=None, offer=None, sk
 
 
 def show_vm_image(cmd, urn=None, publisher=None, offer=None, sku=None, version=None, location=None):
+    from azure.cli.core.azclierror import AzCLIErrorType, AzCLIError
     from azure.cli.core.commands.parameters import get_one_of_subscription_locations
-    usage_err = 'usage error: --plan STRING --offer STRING --publish STRING --version STRING | --urn STRING'
+
     location = location or get_one_of_subscription_locations(cmd.cli_ctx)
     if urn:
         if any([publisher, offer, sku, version]):
-            raise CLIError(usage_err)
+            error_msg = 'If using --urn, do not specify any of --publisher, --offer, --sku, --version'
+            az_error = AzCLIError(AzCLIErrorType.UsageError, error_msg)
+            az_error.set_recommendation("Try to use --urn publisher:offer:sku:version only")
+            raise az_error
+        items = urn.split(":")
+        if len(items) != 4:
+            error_msg = '--urn should be in the format of publisher:offer:sku:version'
+            raise AzCLIError(AzCLIErrorType.UsageError, error_msg)
         publisher, offer, sku, version = urn.split(":")
         if version.lower() == 'latest':
             version = _get_latest_image_version(cmd.cli_ctx, location, publisher, offer, sku)
     elif not publisher or not offer or not sku or not version:
-        raise CLIError(usage_err)
+        error_msg = 'If not using --urn, all of --publisher, --offer, --sku, --version should be specified'
+        raise AzCLIError(AzCLIErrorType.UsageError, error_msg)
     client = _compute_client_factory(cmd.cli_ctx)
     return client.virtual_machine_images.get(location, publisher, offer, sku, version)
 
 
 def accept_market_ordering_terms(cmd, urn=None, publisher=None, offer=None, plan=None):
+    from azure.cli.core.azclierror import AzCLIErrorType, AzCLIError
     from azure.mgmt.marketplaceordering import MarketplaceOrderingAgreements
 
-    usage_err = 'usage error: --plan STRING --offer STRING --publish STRING |--urn STRING'
     if urn:
         if any([publisher, offer, plan]):
-            raise CLIError(usage_err)
-        publisher, offer, _, _ = urn.split(':')
+            error_msg = 'If using --urn, do not specify any of --plan, --offer, --publish'
```

Comment:
Could we extra a error message template for this cases; either one param, or a group of splitted params, others just use it by filling in param name?